### PR TITLE
SAS-43 #manage team assets

### PR DESCRIPTION
**Pull request by Chen2908**
Written by Yaar and me.
UC 6.1- manage team assets- which includes add, edit and remove.
A TeamAsset is an interface which is implemented by Player, Facility and Coach.

### CODE REVIEW

File: src/main/java/com/SAS/User/Player.java [hunk 1]
```diff
@@ -111,4 +114,28 @@ public Team getTeam() {
...
+
+    /**
+     * This function edits the player details
+     * @param details
+     */
+    @Override
+    public void editDetails(List<String> details) {
+        //first is dateOfBirth, second is fieldRole
+        setDateOfBirth(LocalDate.parse(details.get(0)));
+        setFieldRole(convertStringToFieldRole(details.get(1)));
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/User/Player.java]: convertStringToFieldRole can return null. maybe you need to check it before you set the field role.

File: src/main/java/com/SAS/stadium/Stadium.java [hunk 2]
```diff
@@ -81,15 +99,69 @@ public void setGamesList(List<Game> gamesList) {
...
+    /**
+     * This functions edits the stadiums details
+     * @param details
+     */
+    @Override
+    public void editDetails(List<String> details) {
+        //first is name, second is location, third is type
+        setName(details.get(0));
+        setLocation(details.get(1));
+        setFacilityType(convertStringToFacilityType(details.get(2)));
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/stadium/Stadium.java]: Same thing with null here in setFacilityType.

File: src/main/java/com/SAS/teamManagenemt/TeamManagement.java [hunk 3]
```diff
@@ -18,6 +21,52 @@ public TeamManagement(UserController userController) {
...
+     * @param assetType
+     * @param team
+     * @return the created asset
+     */
+    public TeamAsset AddAssetToTeam (String assetType, Team team) {
+        TeamAsset asset = null;
+        //create the asset
+        switch (assetType) {
+            case "Player":
+                asset = (Player) userController.createUser(null, null, null, UserType.PLAYER, true);
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/teamManagenemt/TeamManagement.java:35]: You create the player with null fields and later his details will be edited?

File: src/main/java/com/SAS/teamManagenemt/TeamManagement.java [hunk 4]
```diff
@@ -18,6 +21,52 @@ public TeamManagement(UserController userController) {
...
+                team.addPlayerToTeam((Player)asset);
+                break;
+            case "Facility":
+                asset = new Stadium();
+//                team.addFacility((Facility)asset);
+                break;
+            case "Coach":
+                asset = (Coach) userController.createUser(null, null, null, UserType.COACH, true);
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/teamManagenemt/TeamManagement.java:43]: Same as a player.

File: src/main/java/com/SAS/teamManagenemt/TeamManagement.java [hunk 5]
```diff
@@ -32,14 +81,63 @@ public User addAdditionalTeamOwner(User newTeamOwner, Team team, User nominatedB
...
+     * @return newTeamManager - the team manager
+     */
+    public User addTeamManager(User newTeamManager, Team team, User nominatedBy, boolean approval){
+        //check if the team already has a manager
+        if(teamHasManager(team)){
+            System.out.println("The team already has a manager, please remove it first to nominate a new manager");
+            return newTeamManager;
+        }
+        //check if the user can manage this team and that the user that nominates him is the team owner
+        if (validateTeamManager(newTeamManager, team) && ownsTeam(team, nominatedBy)){
```
Comment (AmitNachimovitz) [inline src/main/java/com/SAS/teamManagenemt/TeamManagement.java]: Maybe add a success message so the user will get a response from the system.

File: src/main/java/com/SAS/teamManagenemt/TeamManagement.java (hunk 3 above)
Comment (Chen2908) [reply src/main/java/com/SAS/teamManagenemt/TeamManagement.java:35]: yes

File: src/main/java/com/SAS/User/Player.java (hunk 1 above)
Comment (Chen2908) [reply src/main/java/com/SAS/User/Player.java]: ok, will check it

**Comment by MrBahur** [comment]
is this PR still relevant? if so, can you please commit the changes requested by Amit?

**Review by MrBahur — CHANGES_REQUESTED** [review]
please comment on the requests by amit, and update the PR or create new PR and close this one.

**Pull request merged**